### DISCUSSION TITLE
Scope of this repo?

### DISCUSSION THREAD

**Discussion by dumblob** [vincenzopalazzo/benchmarks#4] [General]
Is the goal of this repo to only perform Google benchmarks or rather "benchmark V in any imaginable way to find bottle necks, their culprits, and (potential) solutions"?

If the latter, I would file an issue to investigate the non-performance of `-prod` builds on [kostya's machine](https://github.com/kostya/benchmarks/#benchb ) (or in other words why Go is as fast as C on that machine and not on some other machines as discussed in https://github.com/vlang/v/discussions/14222#discussioncomment-2732867 ).

Could the scope of this repo be documented in readme?

**Comment by vincenzopalazzo** [comment]
>Google benchmarks 

Google benchmark is the minimum common divisor between languages because you can use a single library for all the algorithms written in different languages, this avoid also to implement a very stupid benchmark with "end - start" time, but there are well tested euristic implemented by the lib.

>If the latter, I would file an issue to investigate the non-performance of -prod builds on [kostya's machine](https://github.com/kostya/benchmarks/#benchb) (or in other words why Go is as fast as C on that machine and not on some other machines as discussed in https://github.com/vlang/v/discussions/14222#discussioncomment-2732867 ).

Run the benchmark with `-prod` can be an option too

> Could the scope of this repo be documented in readme?

The scope of the readme is all the useful benchmarks written with a good style and how a real code is written between languages.

I don't want include unuseful example, or only language specific example! So if this is your goal I think this repo is not for you, otherwise contributors are welcome, but the code need to be well written

**Reply by dumblob** [reply]
I am asking mainly whether this repo is meant to serve **also** as discussion platform for investigations of **why** foreign benchmarks related to *V* produce results which are not favourable, which are surprising, which are weird, etc.

For that there is no need to contribute any code to this repository altogether. I am really just seeking a public place to discuss **existing** benchmark results and related issues from other places.

Is this in scope of this repository?

**Reply by vincenzopalazzo** [reply]
I'm not open to discuss benchmark that i can not run on my machine, you can run these and run you itlsef without trust any fancy blog post

**Reply by vincenzopalazzo** [reply]
and this is the scope of this repo

**Reply by dumblob** [reply]
Than you for clarification of the scope. How do I "close" (set or mark as "solved") this discussion?

> I'm not open to discuss benchmark that i can not run on my machine

Out of curiosity, what for a machine do you have? The linked kostya's benchmarks are widely known and run on any Linux machine (and probably also MacOS, BSD or *nix machines in general).

>you can run these and run you itlsef without trust any fancy blog post

Could you send me a link to the blog post? I could not find any but would like to read what you read :wink:. Thanks!

**Reply by vincenzopalazzo** [reply]
>Out of curiosity, what for a machine do you have? The linked kostya's benchmarks are widely known and run on any Linux machine (and probably also MacOS, BSD or *nix machines in general).

Any machine! it uses a well-known library google benchmark to get the same result on all machine and remove the compile optimization, but this work it still working in progress because take time

>Could you send me a link to the blog post? I could not find any but would like to read what you read wink. Thanks!

I do not know :) but for sure there are blog that talk about fancy fake benchmark
